clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -773,6 +773,58 @@ cir::FuncOp CIRGenModule::getOrCreateCIRFunction(
     StringRef mangledName, mlir::Type funcType, GlobalDecl gd, bool forVTable,
     bool dontDefer, bool isThunk, ForDefinition_t isForDefinition,
     mlir::ArrayAttr extraAttrs) {
+  const Decl *d = gd.getDecl();
+
+  if (isThunk)
+    errorNYI(d->getSourceRange(), "getOrCreateCIRFunction: thunk");
+
+  // In what follows, we continue past 'errorNYI' as if nothing happened because
+  // the rest of the implementation is better than doing nothing.
+
+  // Any attempts to use a MultiVersion function should result in retrieving the
```

Comment:
Is this comment out of place?  Looks like it should be with the actual multiversion stuff on 793.

Reply:
I think you're right. When this comment was originally written (by you, incidentally), line 793 immediately followed line 786, so the comment made sense here. When the OpenMP handling was added, the comment wasn't moved but it should have been. I'll fix that here. The classic codegen will still have this comment location issue.

Reply:
Fixed in classic-codegen: 785ab45c2b300ba6291d8fb21762eabc050275f6